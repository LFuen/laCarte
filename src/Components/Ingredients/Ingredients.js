import React, {Component} from 'react'
import LaCarteContext from '../../context/LaCarteContext'
import {Link} from 'react-router-dom'
import './Ingredients.css'


class Ingredients extends Component{
    
    static defaultProps = {
        match: {
            params: {}
        }
    }
    static contextType = LaCarteContext;

    render(){
        const {meals} = this.context

        if(meals.length > 0){
            const ingID = this.props.match.params.id
            const ingr = meals.length ? meals.find(ing => ing.id === Number(ingID)) : ''
            
            let listing = ingr.ingredients.substring(1)
            listing = listing.slice(0, -1)
            let ingredients = listing.split(',')

            let chefList = ingr.chef.substring(1)
            chefList = chefList.slice(0, -1)
            let chefs = chefList.split(',')

            if (ingr) {
                return (
                    <div className='page'>
                        <h1>{ingr.meal_name}</h1>
                        <img src={ingr.img_url} className='mealIng' alt={ingr.meal_name} title={ingr.meal_name}/>
                        <div className='group'> 
                            <div className='ingredients'>
                                <h2>Ingredients:</h2>
                                {ingredients.map((list, idx) => { return (
                                    <p key={idx} className='listing'>{list}</p>
                                )})}
                            </div>                        
                            <br/>
                            <div className='chefs'>                        
                                <h2>Chefs</h2>             
                                <p className='listing'>Who would You like to prepare your meal?</p>                

                                {chefs.map((chef, idx) => {return (
                                    <label key={idx}>
                                        <input type='radio' value={chef} required/><br/><p className='listing'>{chef}</p>
                                    </label>
                                )})}
                            </div>                        
                        </div>
                        <Link to={`/orders/${ingr.id}`}><button type='submit' className='shadow'>Order</button></Link>
                    </div>
                )
            } else {
                return (<h3>Sorry, we don't cook that yet!</h3>)
            }
        } else {
            return (<h3> Loading... </h3>)
        }
    }
}

export default Ingredients

