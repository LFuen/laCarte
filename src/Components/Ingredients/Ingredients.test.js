import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import ShallowRenderer from 'react-test-renderer/shallow'
import {BrowserRouter} from 'react-router-dom'
import LaCarteContext from '../../context/LaCarteContext'
import Ingredients from './Ingredients'



describe(`Ingredients Component`, () => {
    describe(`Snapshot`, () => {
        it(`renders the UI as expected`, () => {
            const ingredients = renderer.create(<BrowserRouter><Ingredients/></BrowserRouter>)
                expect(ingredients).toMatchSnapshot()
        })
    })

    describe(`Smoke test`, () => {
        it(`Renders without crashing`, () => {
            
            const div = document.createElement('div');
            const renderer = new ShallowRenderer()

            const value = {
                meals: 'string',
                chefs: 'string',
                cuisines: 'string',
                orders: 'string',
                addOrder: 'string',
                updateOrder: 'string',
                deleteOrder: 'string',
            };

            renderer.render(
                <BrowserRouter>
                    <LaCarteContext.Provider value={value}>
                        <Ingredients />
                    </LaCarteContext.Provider>                
                </BrowserRouter>,
                div)
            ReactDOM.unmountComponentAtNode(div)
        })
    })
})