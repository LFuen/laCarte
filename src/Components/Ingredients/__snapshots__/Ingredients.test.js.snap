// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Ingredients Component Snapshot renders the UI as expected 1`] = `
<h3>
   Loading... 
</h3>
`;
